src/prefect/server/utilities/messaging/memory.py

```diff
@@ -26,6 +30,24 @@
 
 logger: "logging.Logger" = get_logger(__name__)
 
+# Simple global counters by topic with thread-safe access
+_metrics_lock: asyncio.Lock | None = None
```

Comment:
If you want this to be thread-safe, you'll need this to be a `threading.Lock` instead. I don't think it needs to be thread-safe though, this is mostly a reaction to the comment.